app/uk/gov/hmrc/teamsandrepositories/controller/WebhookController.scala

```diff
@@ -48,16 +48,21 @@ class WebhookController @Inject()(
       }
     }
 
-    def deletedRepositoryEvent(repoName: String): Future[Unit] = {
+    def deletedRepositoryEvent(repoName: String): Future[Unit] =
       repositoriesPersistence.findRepo(repoName).flatMap {
         case Some(repo) =>
           for {
-            _   <- deletedRepositoriesPersistence.set(Seq(DeletedGitRepository.fromGitRepository(repo, Instant.now())))
+            _   <- deletedRepositoriesPersistence.set(DeletedGitRepository.fromGitRepository(repo, Instant.now()))
             _   <- persistingService.repositoryDeleted(repoName)
           } yield ()
-        case None => deletedRepositoriesPersistence.set(Seq(DeletedGitRepository(repoName, Instant.now())))
+        case None =>
+          deletedRepositoriesPersistence.set(DeletedGitRepository(repoName, Instant.now()))
+      }.recover {
+        case ex: com.mongodb.MongoWriteException if ex.getMessage.contains("duplicate key error") =>
```

Comment:
you can case match on [DuplicateKey](https://github.com/hmrc/hmrc-mongo/blob/a4eb51dd8f7b809903724d66932788256aeac274/hmrc-mongo-common/src/main/scala/uk/gov/hmrc/mongo/MongoUtils.scala#L233-L235) instead